paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java

```diff
@@ -801,6 +803,86 @@ public boolean dropItem(boolean dropAll) {
         // Paper end - Fix HumanEntity#drop not updating the client inv
     }
 
+    // Paper start - Extend HumanEntity#dropItem API
+    @Override
+    public @Nullable org.bukkit.entity.Item dropItem(final @NotNull ItemStack itemStack) {
+        return this.dropItem(itemStack, null, false);
+    }
+
+    @Override
+    public @Nullable org.bukkit.entity.Item dropItem(final @NotNull ItemStack itemStack, final @Nullable java.util.UUID thrower) {
+        return this.dropItem(itemStack, thrower, false);
+    }
+
+    @Override
+    public @Nullable org.bukkit.entity.Item dropItem(final @NotNull ItemStack itemStack, final @Nullable java.util.UUID thrower, final boolean throwRandomly) {
+        final int slot = this.inventory.first(itemStack);
+        if (slot == -1) {
+            return null;
+        }
+
+        return this.dropItem(slot, thrower, throwRandomly);
+    }
+
+    @Override
+    public @Nullable org.bukkit.entity.Item dropItem(final int slot) {
+        return this.dropItem(slot, null, false);
+    }
+
+    @Override
+    public @Nullable org.bukkit.entity.Item dropItem(final int slot, final @Nullable java.util.UUID thrower) {
+        return this.dropItem(slot, thrower, false);
+    }
+
+    @Override
+    public @Nullable org.bukkit.entity.Item dropItem(final int slot, final @Nullable java.util.UUID thrower, final boolean throwRandomly) {
+        // Make sure the slot is in bounds
+        if (slot < 0 || this.inventory.getSize() <= slot) {
```

Comment:
```suggestion
        if (slot < 0 || slot >= this.inventory.getSize()) {
```
Turning it around makes it clearer